### PR TITLE
Add Dockerfile, script to deploy to dockerhub, CI job to deploy to docker

### PR DESCRIPTION
**Pull request by giacomolicari**
The PR adds the following features:
- Dockerfile containing a nodejs image which builds the project and serves it
- script to deploy the docker image to dockerhub
- job on the CI to deploy to docker

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -27,3 +27,24 @@ jobs:
...
+    - uses: actions/checkout@v2
+    - name: Dockerhub login
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USER }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    - name: Deploy Main
+      if: github.ref == 'refs/heads/master'
+      run: bash docker/deploy.sh staging
+    - name: Deploy Develop
```
Comment (fleupold) [inline .github/workflows/CI.yml:45]: nit: This repo doesn't have a develop branch (we can remove this arm).

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -27,3 +27,24 @@ jobs:
...
+      DOCKERHUB_PROJECT: gp-v2-trading-bot
+    steps:
+    - uses: actions/checkout@v2
+    - name: Dockerhub login
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USER }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    - name: Deploy Main
+      if: github.ref == 'refs/heads/master'
```
Comment (fleupold) [inline .github/workflows/CI.yml:43]: master is called main in this repo

File: docker/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+if [ "$1" = "develop" -o "$1" = "main" ]; then # if using other branches, change branch names accordingly.
+    # If image does not exist, don't use cache
+    docker pull gnosispm/$DOCKERHUB_PROJECT:$1 && \
+    docker build -t $DOCKERHUB_PROJECT -f docker/Dockerfile . --cache-from gnosispm/$DOCKERHUB_PROJECT:$1 || \
+    docker build -t $DOCKERHUB_PROJECT -f docker/Dockerfile .
+else
+    docker pull gnosispm/$DOCKERHUB_PROJECT:staging && \
+    docker build -t $DOCKERHUB_PROJECT -f docker/Dockerfile . --cache-from gnosispm/$DOCKERHUB_PROJECT:staging || \
+    docker build -t $DOCKERHUB_PROJECT -f docker/Dockerfile .
```
Comment (fleupold) [inline docker/deploy.sh:14]: What's the difference between the else branch and the first if branch? Shouldn't we always just build based on $1 and use --cache-from staging?

**Review by fleupold — APPROVED** [review]
Thanks a lot for taking care of this!

master is called main in this repo, other comments are not blocking merge.

File: docker/deploy.sh (hunk 3 above)
Comment (giacomolicari) [reply docker/deploy.sh:14]: yes, we can remove the second if

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FHw8vYF4DNRCKY%2Fgiphy.gif%3Fcid%3D2972a2b15d1d81cf6d6d7963363a4f7b%26rid%3Dgiphy.gif&token=shipit" data-gitmeme-token="shipit"></a>

**Pull request merged**